scene/resources/particle_process_material.cpp

```diff
@@ -683,11 +683,13 @@ void ParticleProcessMaterial::_update_shader() {
 		code += "	pos.z = 0.0;\n";
 	}
 	code += "	// apply linear acceleration\n";
-	code += "	force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * mix(linear_accel_min, linear_accel_max, rand_from_seed(alt_seed)) : vec3(0.0);\n";
+	code += "	vec3 linear_accel_rand = mix(linear_accel_min, linear_accel_max, rand_from_seed(alt_seed));\n";
+	code += "	force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * linear_accel_rand : vec3(0.0);\n";
```

Comment:
```suggestion
	code += "	float linear_accel_rand = rand_from_seed(alt_seed);\n";
	code += "	force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * mix(linear_accel_min, linear_accel_max, linear_accel_rand) : vec3(0.0);\n";
```
To do the least processing in that branch